File: tests/fixtures/language_adapter_symbols/sample.clj
;; FORMAT-DERIVED: clojure.org/reference/reader ("Comments" -- ; line comments only, no block
;; comment form; "Discard" -- #_ elides the next form; "Characters" -- bare \c / \newline / \space;
;; "Strings") and clojure.org/reference/special_forms plus core macros def/defn/defprotocol/
;; defrecord -- this fixture follows those grammar productions, not this repo's extractor.

(ns sample.core)

(defn greet
  "Returns a greeting string."
  [name]
  (str "hello " name))

;; #_ elides the entire next form -- this defn must never become a symbol.
#_(defn unused-fn [x] x)

(def max-width 80)

(defprotocol Shape
  (area [this]))

(defrecord Point [x y]
  Shape
  (area [_] 0))

(defmulti describe :kind)

(defmethod describe :circle [_] "a circle")

(def line-char \-)
(def newline-char \newline)
(def greeting-text "a \"quoted\" word")

File: tests/fixtures/language_adapter_symbols/sample.fs
(* FORMAT-DERIVED: The F# Language Specification (F# 4.1), section 3.1 "Lexical Analysis"
   (comments, string/verbatim/triple-quoted string literals) and section 10 "Namespaces and
   Modules" / section 8 "Type Definitions" / section 12 "Exception Definitions" for
   `namespace`/`module`/`type`/`exception`. No live internet access was used to fetch the current
   published spec while writing this fixture -- these are the commonly-cited F# 4.1 section
   numbers, and the fixture follows those grammar productions, not this repo's extractor. *)

namespace Sample

/// Greets someone by name.
let greet name = "hello, " + name

type Shape =
  | Circle of float
  | Rectangle of float * float

exception BadShape of string

module Inner =
  let id x = x
